metrics/ci_integrations/test/integration/ci/sync_stage/builds/new_builds_sync_stage_test.dart

```diff
@@ -0,0 +1,495 @@
+      ".call() does not continue the sync if an error occurs while fetching the builds after the fetched last build",
+      () async {
+        whenFetchBuildsAfter().thenAnswer((_) => Future.error(error));
+
+        await newBuildsSyncStage.call(syncConfig);
+
+        verifyNever(sourceClient.fetchCoverage(any));
+        verifyNever(destinationClient.addBuilds(any, any));
+      },
+    );
+
+    test(
+      ".call() returns an error if an error occurs while fetching the builds after the fetched last build",
+      () async {
+        whenFetchBuildsAfter().thenAnswer((_) => Future.error(error));
+
+        final result = await newBuildsSyncStage.call(syncConfig);
+
+        expect(result.isError, isTrue);
+      },
+    );
+
+    test(
+      ".call() does not continue the sync if the fetched builds after are empty",
+      () async {
+        whenFetchBuildsAfter().thenAnswer((_) => Future.value([]));
+
+        await newBuildsSyncStage.call(syncConfig);
+
+        verifyNever(sourceClient.fetchCoverage(any));
+        verifyNever(destinationClient.addBuilds(any, any));
+      },
+    );
+
+    test(
+      ".call() returns a success if the fetched builds after are empty",
+      () async {
+        whenFetchBuildsAfter().thenAnswer((_) => Future.value([]));
+
+        final result = await newBuildsSyncStage.call(syncConfig);
+
+        expect(result.isSuccess, isTrue);
+      },
+    );
+
+    test(
+      ".call() does not fetch the coverage for the fetched builds after if the coverage option is not enabled in the sync config",
+      () async {
+        whenFetchBuildsAfter().thenAnswer((_) => Future.value(builds));
+
+        await newBuildsSyncStage.call(syncConfig);
+
+        verifyNever(sourceClient.fetchCoverage(build));
+      },
+    );
+
+    test(
+      ".call() adds the fetched builds after to the destination if the coverage option is not enabled in the sync config",
+      () async {
+        whenFetchBuildsAfter().thenAnswer((_) => Future.value(builds));
+
+        await newBuildsSyncStage.call(syncConfig);
+
+        verify(
+          destinationClient.addBuilds(destinationProjectId, builds),
+        ).called(once);
+      },
+    );
+
+    test(
+      ".call() fetches coverage for the fetched builds after if the coverage option is enabled in the sync config",
+      () async {
+        whenFetchBuildsAfter().thenAnswer((_) => Future.value(builds));
+
+        await newBuildsSyncStage.call(syncConfigWithCoverage);
+
+        verify(sourceClient.fetchCoverage(build)).called(once);
+        verify(sourceClient.fetchCoverage(anotherBuild)).called(once);
+      },
+    );
+
+    test(
+      ".call() fetches coverage for the fetched builds after if the coverage option is enabled in the sync config",
+      () async {
+        whenFetchBuildsAfter().thenAnswer((_) => Future.value(builds));
+
+        await newBuildsSyncStage.call(syncConfigWithCoverage);
+
+        verify(sourceClient.fetchCoverage(build)).called(once);
```

Comment:
```suggestion
        verify(sourceClient.fetchCoverage(argThat(oneOf(builds)))).called(equals(builds.length));
```
What do you think about replacing similar calls with something like the above?